serpent-moltres/fuel-column-sensitivity/postprocessing.py

```diff
@@ -0,0 +1,406 @@
+    lim18 = [2, 11, 18]  # from 18 to 3
+    lim15 = [2, 10, 15]  # from 15 to 3
+    lim12 = [1, 7, 12]  # from 12 to 3
+    lim9 = [1, 5, 9]  # from 9 to 3
+    lim6 = [1, 3, 6]  # from 6 to 3
+    lim3 = [1, 2, 3]  # from 3 to 3
+
+    e = np.zeros((7, 2, 3))
+    file1 = '3D-assembly-noLBP-' + str(temp) + '-26G_axial_0002.csv'
+    file2 = '3D-assembly-noLBP-' + str(temp) + '-21G_axial_0002.csv'
+    e[6] = np.array(calc_error(file1, file2, lim26, lim21))
+    file2 = '3D-assembly-noLBP-' + str(temp) + '-18G_axial_0002.csv'
+    e[5] = np.array(calc_error(file1, file2, lim26, lim18))
+    file2 = '3D-assembly-noLBP-' + str(temp) + '-15G_axial_0002.csv'
+    e[4] = np.array(calc_error(file1, file2, lim26, lim15))
+    file2 = '3D-assembly-noLBP-' + str(temp) + '-12G_axial_0002.csv'
+    e[3] = np.array(calc_error(file1, file2, lim26, lim12))
+    file2 = '3D-assembly-noLBP-' + str(temp) + '-9G_axial_0002.csv'
+    e[2] = np.array(calc_error(file1, file2, lim26, lim9))
+    file2 = '3D-assembly-noLBP-' + str(temp) + '-6G_axial_0002.csv'
+    e[1] = np.array(calc_error(file1, file2, lim26, lim6))
+    file2 = '3D-assembly-noLBP-' + str(temp) + '-3G_axial_0002.csv'
+    e[0] = np.array(calc_error(file1, file2, lim26, lim3))
+
+    xticks = [3, 6, 9, 12, 15, 18, 21]
+    xlabel = 'Number of energy groups'
+    plot_error_acc_study(e, 'noLBP-' + str(temp) + '-er-final', xticks, xlabel)
+    return
+
+
+def LBP_L2error(temp):
+    '''
+    This function plots the L2 norm relative error for the different
+    group structures for the LBP cases.
+
+    Parameters:
+    -----------
+    temp: [float]
+        temperature in K: 600 or 1200K
+    '''
+
+    lim26 = [4, 16, 26]  # from 26 to 3
+    lim21 = [2, 12, 21]  # from 21 to 3
+    lim18c = [2, 9, 18]  # from 18 to 3
+    lim15c = [2, 7, 15]  # from 15 to 3
+    lim12 = [1, 7, 12]  # from 12 to 3
+    lim9 = [1, 5, 9]  # from 9 to 3
+    lim6 = [1, 3, 6]  # from 6 to 3
+    lim3 = [1, 2, 3]  # from 3 to 3
+
+    e = np.zeros((7, 2, 3))
+    file1 = '3D-assembly-LBP-' + str(temp) + '-26G_axial_0002.csv'
+    file2 = '3D-assembly-LBP-' + str(temp) + '-21G_axial_0002.csv'
+    e[6] = np.array(calc_error(file1, file2, lim26, lim21))
+    file2 = '3D-assembly-LBP-' + str(temp) + '-18Gc_axial_0002.csv'
+    e[5] = np.array(calc_error(file1, file2, lim26, lim18c))
+    file2 = '3D-assembly-LBP-' + str(temp) + '-15Gc_axial_0002.csv'
+    e[4] = np.array(calc_error(file1, file2, lim26, lim15c))
+    file2 = '3D-assembly-LBP-' + str(temp) + '-12G_axial_0002.csv'
+    e[3] = np.array(calc_error(file1, file2, lim26, lim12))
+    file2 = '3D-assembly-LBP-' + str(temp) + '-9G_axial_0002.csv'
+    e[2] = np.array(calc_error(file1, file2, lim26, lim9))
+    file2 = '3D-assembly-LBP-' + str(temp) + '-6G_axial_0002.csv'
+    e[1] = np.array(calc_error(file1, file2, lim26, lim6))
+    file2 = '3D-assembly-LBP-' + str(temp) + '-3G_axial_0002.csv'
+    e[0] = np.array(calc_error(file1, file2, lim26, lim3))
```

Comment:
Same comments as before (because it's the same code.... ) except now, I recomend this should also be its own function, which returns the objects you need (like e).